### PR TITLE
Check unfinished persons before import

### PR DESCRIPTION
**Pull request by Baiqiang**
This is @SAuroux 's idea. If someone clicked the `import_persons` too early, it will show an alert to prevent it.
image

### CODE REVIEW

File: webroot/results/admin/scripts/import_persons.php [hunk 1]
```diff
@@ -8,9 +8,21 @@
 // All are done in persons_finish_unfinished_ACTION.php
 // See https://github.com/thewca/worldcubeassociation.org/issues/2286
 
+// Check if any persons with numeric ID
+$compId = mysqlEscape( $compId );
+$unfinishedPersons = dbValue("
```
Comment (jfly) [inline webroot/results/admin/scripts/import_persons.php]: nit: By the naming convention we've been following, `unfinishedPersons` would actually be an array of unfinished persons. Instead, we'd name this `unfinishedPersonsCount`, or `numberOfUnfinishedPersons` if you prefer =)

File: webroot/results/admin/scripts/remove_imported_data.php [hunk 2]
```diff
@@ -19,4 +19,4 @@
 //     array('s', &$compId)
 //   );
 
-print "Removal attempted.";
+jsonReturn();
```
Comment (jfly) [inline webroot/results/admin/scripts/remove_imported_data.php]: So these messages ("Imported Results..." and "Removal attempted.") were never actually used anywhere?

File: webroot/results/js/data_upload_help.js [hunk 3]
```diff
@@ -44,8 +44,12 @@ function get_comp_info(obj) {
       //Load page into table cell.
       $.ajax({
         url: $(this).attr('href'),
+        dataType: 'json'
```
Comment (jfly) [inline webroot/results/js/data_upload_help.js]: nit: missing trailing comma here

File: webroot/results/js/data_upload_help.js [hunk 4]
```diff
@@ -44,8 +44,12 @@ function get_comp_info(obj) {
...
       })
       .done( function(data) {
+        if (data.status != 'OK') {
+          alert(data.message)
```
Comment (jfly) [inline webroot/results/js/data_upload_help.js]: nit: missing semicolon here

File: webroot/results/admin/upload_results.php [hunk 5]
```diff
@@ -7,7 +7,7 @@
 adminHeadline('Upload Competition Results &amp; Scrambles');
 
 $scripts = new WCAClasses\WCAScripts();
-$scripts->add('data_upload_help.js');
+$scripts->add('data_upload_help.js?ver=201802082346');
```
Comment (jfly) [inline webroot/results/admin/upload_results.php]: Won't this result in two url parameters (both `v=` and `ver=`?) Would the solution instead be to bump the version in [`webroot/results/includes/WCAClasses/WCAScripts.class.php`](https://github.com/jfly/worldcubeassociation.org/blob/285c4a745f20ffd5537b10e3a1f8d925c0877ae0/webroot/results/includes/WCAClasses/WCAScripts.class.php#L10)?

I'm surprised we even need this cache busting technique, I checked and it looks like our server isn't aggressively caching this stuff:

image

But whatever, let's not waste too much time improving this ancient code =)

File: webroot/results/admin/scripts/remove_imported_data.php (hunk 2 above)
Comment (Baiqiang) [reply webroot/results/admin/scripts/remove_imported_data.php]: No. We don't use it.

**Comment by Baiqiang** [comment]
I've squashed all commits into one. Going to merge this after travis passing the test.

**Pull request merged**